aiida/backends/tests/test_nodes.py

```diff
@@ -1702,100 +1621,385 @@ def test_deletion_non_existing_pk(self):
+              |  |                 v                v  v
+            +------+ input_calc  +----+  create   +------+
+            |  DI  | ----------> | C0 | --------> |  DO  |
+            +------+             +----+           +------+
         """
-        in1, in2, wf, slave1, outp1, outp2, slave2, outp3, outp4 = self._create_calls_n_returns_graph()
-        # The inputs are not harmed.
-        uuids_check_existence = (in1.uuid, in2.uuid)
-        # the slaves and their outputs have to disappear since calls are followed!
-        uuids_check_deleted = [n.uuid for n in (wf, slave1, outp1, outp2, outp3, slave2, outp4)]
 
-        with Capturing():
-            delete_nodes([wf.pk], verbosity=2, force=True, follow_calls=True, follow_returns=True)
+        data_i = orm.Data().store()
+        data_o = orm.Data().store()
 
-        self._check_existence(uuids_check_existence, uuids_check_deleted)
+        calc_0 = orm.CalculationNode()
+        work_1 = orm.WorkflowNode()
+        work_2 = orm.WorkflowNode()
+
+        calc_0.add_incoming(data_i, link_type=LinkType.INPUT_CALC, link_label='inpcalc')
+        work_1.add_incoming(data_i, link_type=LinkType.INPUT_WORK, link_label='inpwork')
+        calc_0.add_incoming(work_1, link_type=LinkType.CALL_CALC, link_label='callcalc')
+        work_1.add_incoming(work_2, link_type=LinkType.CALL_WORK, link_label='callwork')
 
-    def test_deletion_with_calls_no_returns(self):
+        work_2.store()
+        work_1.store()
+        calc_0.store()
+
+        data_o.add_incoming(calc_0, link_type=LinkType.CREATE, link_label='create0')
+        data_o.add_incoming(work_1, link_type=LinkType.RETURN, link_label='return1')
+        data_o.add_incoming(work_2, link_type=LinkType.RETURN, link_label='return2')
+
+        return data_i, data_o, calc_0, work_1, work_2
+
+    def test_delete_cases(self):
         """
-        Checking the case where I follow calls and not return links for deletion
+        Using a minimal graph (almost only one of each type of link) will test all
+        the conditions established for the consistent deletion of nodes.
         """
-        in1, in2, wf, slave1, outp1, outp2, slave2, outp3, outp4 = self._create_calls_n_returns_graph()
-        # The inputs are not harmed.
-        uuids_check_existence = [n.uuid for n in (in1, in2, outp1, outp3, outp4)]
-        # the slaves and their outputs have to disappear since calls are followed!
-        uuids_check_deleted = [n.uuid for n in (wf, slave1, slave2, outp2)]
 
+        # By default deleting input data should eliminate everything that it
+        # generated downwards.
+        di, do, c0, w1, w2 = self._create_minimal_graph()
+        uuids_check_existence = [n.uuid for n in ()]
+        uuids_check_deleted = [n.uuid for n in (di, do, c0, w1, w2)]
         with Capturing():
-            delete_nodes([wf.pk], verbosity=2, force=True, follow_calls=True, follow_returns=False)
+            delete_nodes([di.pk], verbosity=2, force=True)
```

Comment:
Remove `verbosity=2` from here and below when capturing output.

Otherwise, put `with Capturing() as output`, then pass `output` as an additional parameter to `_check_existence()` (you'll need to move `_check_existence()` inside the `with` block) and then, if there is an exception, put the output in the exception message.